pkg/appmeshinject/inject.go

```diff
@@ -0,0 +1,90 @@
+package appmeshinject
+
+import (
+	"errors"
+	appmesh "github.com/aws/aws-app-mesh-controller-for-k8s/apis/appmesh/v1beta2"
+	corev1 "k8s.io/api/core/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+const (
+	ecrSecret = "appmesh-ecr-secret"
+	// We don't want to make this configurable since users shouldn't rely on this
+	// feature to set a fsGroup for them. This feature is just to protect innocent
+	// users that are not aware of the limitation of iam-for-service-accounts:
+	// https://github.com/aws/amazon-eks-pod-identity-webhook/issues/8
+	// Users should set fsGroup on the pod spec directly if a specific fsGroup is desired.
+	defaultFSGroup int64 = 1337
+)
+
+var (
+	config    Config
+	injectLog = ctrl.Log.WithName("appmesh_inject")
+)
+
+type PodMutator interface {
+	mutate(*corev1.Pod) error
+}
+
+func New(cfg Config) {
+	config = cfg
+}
+
+type AppMeshCNIMutator struct{}
```

Comment:
seems cni mutator don't belong here..(it's should be it's own file)

Reply:
ok